### PR TITLE
Feat:Cart

### PR DESCRIPTION
**Pull request by rehanmohiuddin**
- Added Cart Page
- Cart Reducers
- Cart Actions
- Cart Functions & Mappers
- Preview Can Be Seen Below Netlify Link

### CODE REVIEW

File: src/Frontend/Pages/Products/Products.jsx [hunk 1]
```diff
@@ -48,8 +60,13 @@ function Products() {
...
               </div>
               <div className="product-actions">
-                <Button type="secondary" data="Add To Cart" />
-                <Button type="orange" data="Buy Now" />
+                <Button type="secondary" data="Add To Wishlist" />
+                <Button
+                  navigate={null}
+                  clickFun={() => handleAddToCart(_product)}
+                  type="orange"
+                  data="Add To Cart"
```
Comment (shivanigangadharan) [inline src/Frontend/Pages/Products/Products.jsx:68]: After adding to cart, this button's text can be changed to something like "Added".

File: src/Frontend/Pages/Products/Products.jsx [hunk 2]
```diff
@@ -16,6 +21,13 @@ function Products() {
   useEffect(() => setProducts(filteredProducts), [filteredProducts]);
   useEffect(() => setProducts(products), [products]);
 
+  const handleAddToCart = (product) => {
+    cart.dispatch({ type: ADD_TO_CART, data: { product } });
```
Comment (shivanigangadharan) [inline src/Frontend/Pages/Products/Products.jsx:25]: Need to fix : If a product is already in the cart, clicking on add to cart keeps increasing the count on the cart badge in navbar. Then the cart also shows the same item 5 times.

**Review by shivanigangadharan — COMMENTED** [review]
Requires a small fix, other than that code and UI is good. Nice work.

**Review by ArjunGTX — COMMENTED** [review]
A confusion regarding the UX, I am able to add items to cart even without loginng in. That shouldn't be the case I guess. Cart and wishlist are supposed to be protected routes.

File: src/Frontend/Pages/Products/Products.jsx (hunk 1 above)
Comment (rehanmohiuddin) [reply src/Frontend/Pages/Products/Products.jsx:68]: I have Implemented Snackbar for that you can check in deploy preview

File: src/Frontend/Pages/Products/Products.jsx (hunk 2 above)
Comment (rehanmohiuddin) [reply src/Frontend/Pages/Products/Products.jsx:25]: Missed the edge case thanks for noticing will fix this

**Pull request merged**